signals/weather.py

```
@@ -0,0 +1,256 @@
+    def display_name(self) -> str:
+        return "Weather Revenue Impact"
+
+    def collect(self) -> SignalResult:
+        """Fetch, enrich, dedup, and build weather card sections.
+
+        Never raises — all exceptions are caught and returned as ERROR results.
+        """
+        try:
+            return self._collect_inner()
+        except Exception as exc:
+            logger.exception("WeatherSignal.collect() failed: %s", exc)
```

Ruff (0.15.2)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)